astropy/nddata/tests/test_nduncertainty.py

```diff
@@ -354,3 +364,24 @@ def test_assigning_uncertainty_with_bad_unit_to_parent_fails(NDClass,
     v = UncertClass([1, 1], unit=u.second)
     with pytest.raises(u.UnitConversionError):
         ndd.uncertainty = v
+
+
+@pytest.mark.parametrize(
+    'UncertClass', uncertainty_types_with_conversion_support
+)
+def test_self_conversion_via_variance_supported(UncertClass):
+    uncert = np.arange(1, 11).reshape(2, 5) * u.adu
+    start_uncert = UncertClass(uncert)
+    final_uncert = start_uncert.represent_as(UncertClass)
+    assert_array_equal(start_uncert.array, final_uncert.array)
+    assert start_uncert.unit == final_uncert.unit
```

Comment:
Could you please add a test (either in this test function or a separate one) that checks data values and units for conversion from one uncertainty class to another?

I think the test you have here ensures that the round trip through `VarianceUncertainty` works but in principle the conversion to/from variance could have compensating errors.

Checks for converting to `VarianceUncertainty` from each of the `uncertainty_types_with_conversion_support` should be sufficient, since the test above is then basically testing conversion back.

Reply:
Done